src/components/data/kanban/kanban.tsx

```diff
@@ -142,71 +88,29 @@ export const Kanban: React.FC<KanbanProps> = ({
     }
   };
 
-  const handleObjectChange = (
-    object: AttributeData,
-    sourceColumnIndex: number,
-    targetColumnIndex: number,
-    sourceObjectIndex: number,
-    targetObjectIndex: number,
-  ) =>
-    moveObject(
-      object,
-      sourceColumnIndex,
-      targetColumnIndex,
-      sourceObjectIndex,
-      targetObjectIndex,
-    );
-
-  /**
-   * Moves `object` to column/position at given indices and updates `objectListsState`.
-   * on`onObjectChange` and `onObjectListsChange` are called subsequently.
-   * @param object
-   * @param sourceColumnIndex
-   * @param targetColumnIndex
-   * @param sourceObjectIndex
-   * @param targetObjectIndex
-   */
   const moveObject = (
-    object: AttributeData,
+    object: React.ReactNode,
     sourceColumnIndex: number,
     targetColumnIndex: number,
     sourceObjectIndex: number,
     targetObjectIndex: number,
   ) => {
-    const newObjectListsState = [...objectListsState];
-    const sourceObjectList = newObjectListsState[sourceColumnIndex];
-    const targetObjectList = newObjectListsState[targetColumnIndex] || [];
+    const newComponentListState = [...componentListState];
+    const sourceItems = newComponentListState[sourceColumnIndex].items;
+    const targetItems = newComponentListState[targetColumnIndex].items;
     const _targetObjectIndex =
-      targetObjectIndex === -1 ? targetObjectList.length : targetObjectIndex;
-
-    // Remove item from source column
-    sourceObjectList.splice(sourceObjectIndex, 1);
+      targetObjectIndex === -1 ? targetItems.length : targetObjectIndex;
 
-    // Add item to target column
-    targetObjectList.splice(_targetObjectIndex, 0, object);
+    sourceItems.splice(sourceObjectIndex, 1);
+    targetItems.splice(_targetObjectIndex, 0, object);
 
-    // Update objectListsState
-    setObjectListsState(newObjectListsState);
-
-    // Call callbacks
-    onObjectChange?.(
-      object,
-      sourceColumnIndex,
-      targetColumnIndex,
-      sourceObjectIndex,
-      _targetObjectIndex,
-    );
-    onObjectListsChange?.(newObjectListsState);
+    setComponentListState(newComponentListState);
+    onComponentListChange?.(newComponentListState);
```

Comment:
If I understand this correctly, newComponentListState is of type `KanbanColumn[]`, which has a title and a list of items of type React Node. I am wondering: can the information easily be obtained from `newComponentListState` about what has been changed? Since this information would then have to be passed along to the backend to update the data of the items :thinking: